src/core/interactivity/selection.ts

```diff
@@ -211,30 +211,30 @@ const handleDataPointEvent = (event: ScenegraphEvent, item: Item) => {
         identities = getSelectionIdentitiesFromData(data),
         selection = resolveSelectedIdentities(identities);
     if (selection.length > 0) {
```

Comment:
Thanks for noticing this. I hadn't anticipated that someone would be attempting to work with the current dev branch, as it's not quite ready yet. This unclosed `if` definitely shouldn't be here though and I also had this fix queued up in my next commit.

I'm really keen to onboard your other changes so would it please be possible to revise function so only the `switch` remains, i.e.:

```
const handleDataPointEvent = (event: ScenegraphEvent, item: Item) => {
    const { selectionManager } = hostServices,
        data = resolveDataFromItem(item),
        identities = getSelectionIdentitiesFromData(data),
        selection = resolveSelectedIdentities(identities);
    switch (true) {
        case isSelectionLimitExceeded(selection): {
            dispatchSelectionAborted(true);
            return;
        }
        case selection.length > 0: {
            hideTooltip();
            selectionManager.select(selection);
            store.dispatch(
                updateSelectors(
                    <ISelectionId[]>selectionManager.getSelectionIds()
                )
            );
            return;
        }
        default: {
            clearSelection();
            store.dispatch(
                updateSelectors(
                    <ISelectionId[]>selectionManager.getSelectionIds()
                )
            );
            return;
        }
    }
};
```

This will ensure that the selection manager will handle the cases when no selection IDs are present, as the `default` condition will not fire otherwise, as `selection.length` may be `0` if we can't resolve any selectable data points from the event and will need to clean up the selection manager if this happens :)

Reply:
done